onset_maproom/agronomy.py

```diff
@@ -367,123 +368,93 @@ def soil_plant_water_balance(
+    # Runoff
     if runoff is None:
-        runoff = xr.zeros_like(daily_rain).rename("runoff")
-        runoff.attrs = dict(description="Runoff", units="mm")
-    water_balance = xr.Dataset().merge(runoff)
+        runoff = xr.DataArray(np.zeros(1)).squeeze("dim_0")
     # Compute Effective Precipitation
-    peffective = (daily_rain - water_balance.runoff).rename("peffective")
-    peffective.attrs = dict(description="Effective Precipitation", units="mm")
-    water_balance = water_balance.merge(peffective)
-    # Additional vars and coords
-    et = (et * xr.ones_like(water_balance.peffective)).rename("et")
-    water_balance = water_balance.merge(et)
-    if kc_params is not None:
-        for adim in kc_params.dims:
-            if adim not in water_balance.dims and adim != "kc_periods":
-                water_balance[adim] = kc_params[adim]
-    if p_d is not None:
-        for adim in p_d.dims:
-            if adim not in water_balance.dims and adim != time_coord:
-                water_balance[adim] = p_d[adim]
-    # Intializing sm, drainage, et_crop and et_crop_red
-    # Creating variables and broadcasting everybody
-    soil_moisture = xr.full_like(
-        water_balance.peffective, np.nan
-    ).rename("soil_moisture")
-    soil_moisture.attrs = dict(description="Soil Moisture", units="mm")
-    water_balance = water_balance.merge(soil_moisture)
-    drainage = xr.full_like(
-        water_balance.peffective, np.nan
-    ).rename("drainage")
-    drainage.attrs = dict(description="Drainage", units="mm")
-    water_balance = water_balance.merge(drainage)
-    et_crop = xr.full_like(
-        water_balance.peffective, np.nan
-    ).rename("et_crop")
-    et_crop.attrs = dict(description="Crop Evapotranspiration", units="mm")
-    water_balance = water_balance.merge(et_crop)
-    et_crop_red = xr.full_like(
-        water_balance.peffective, np.nan
-    ).rename("et_crop_red")
-    et_crop_red.attrs = dict(description="Reduced Crop Evapotranspiration", units="mm")
-    water_balance = water_balance.merge(et_crop_red)
-    (water_balance,) = xr.broadcast(water_balance)
-    water_balance["soil_moisture"] = water_balance.soil_moisture.copy()
-    water_balance["drainage"] = water_balance.drainage.copy()
-    water_balance["et_crop"] = water_balance.et_crop.copy()
-    water_balance["et_crop_red"] = water_balance.et_crop_red.copy()
+    peffective = (daily_rain - runoff)
     # Give time dimension to sminit
-    t0 = water_balance["soil_moisture"][time_coord][0] - np.timedelta64(1, 'D')
-    sminit0 = sminit + xr.zeros_like(
-        water_balance["soil_moisture"].isel({time_coord: 0})
-        .expand_dims(dim=time_coord).assign_coords({time_coord: [t0.values]})
+    t0 = peffective[time_coord][0]
+    tinit = t0 - np.timedelta64(1, 'D')
+    sminit = sminit + xr.zeros_like(
+        peffective.isel({time_coord: 0})
+        .expand_dims(dim=time_coord)
+        .assign_coords({time_coord: [tinit.values]})
     )
```

Comment:
Sometimes you multiply by `ones_like(...)`, sometimes you add `zeros_like(...)`. Do you have some difference in meaniing in mind when you choose one over the other? Are they both equivalent to some invocation of `broadcast`? I think using `broacast` might make the intent clearer... although I know how to use `ones_like/zeros_like` but I don't know how to use `broadcast` without looking it up, so maybe not.

Reply:
So there is one case left of * ones_like and another of + zeros_like. I don't think there is a reason to use one or the other, other than the thought of the moment.
And I am not using broadcast because in both those cases, I don't know whether what I am adding those 0s or multiplying those 1s to, whether they are a constant or a DataArray already.